src/core/Microsoft.PowerToys.Settings.UI.Library/SettingsUtils.cs

```diff
@@ -70,24 +70,31 @@ public T GetSettings<T>(string powertoy = DefaultModuleName, string fileName = D
         {
             if (SettingsExists(powertoy, fileName))
             {
-                // Given the file already exists, to deserialize the file and read it's content.
-                T deserializedSettings = GetFile<T>(powertoy, fileName);
-
-                // IF the file needs to be modified, to save the new configurations accordingly.
-                if (deserializedSettings.UpgradeSettingsConfiguration())
+                try
                 {
-                    SaveSettings(deserializedSettings.ToJsonString(), powertoy, fileName);
+                    // Given the file already exists, to deserialize the file and read it's content.
+                    T deserializedSettings = GetFile<T>(powertoy, fileName);
+
+                    // If the file needs to be modified, to save the new configurations accordingly.
+                    if (deserializedSettings.UpgradeSettingsConfiguration())
+                    {
+                        SaveSettings(deserializedSettings.ToJsonString(), powertoy, fileName);
+                    }
+
+                    return deserializedSettings;
                 }
 
-                return deserializedSettings;
-            }
-            else
-            {
-                // If the settings file does not exist, to create a new object with default parameters and save it to a newly created settings file.
-                T newSettingsItem = new T();
-                SaveSettings(newSettingsItem.ToJsonString(), powertoy, fileName);
-                return newSettingsItem;
+                // If there are any deserialization issues, log the error and create a new settings.json file.
```

Comment:
We should link to the issue in this comment so that anyone looking at this code knows what kind of "deserialization issue" could occur. Without there wouldn't be enough context for why this code is needed and it might get removed again accidentally.

Reply:
Sounds good. Will do that.